v2/pkg/controller/mpi_job_controller.go

```diff
@@ -687,6 +691,18 @@ func (c *MPIJobController) getRunningWorkerPods(mpiJob *kubeflow.MPIJob) ([]*cor
 	return podList, nil
 }
 
+func (c *MPIJobController) countReadyWorkerPods(workers []*corev1.Pod) (int) {
+	ready := 0
+	for _, pod := range workers {
+		for _, c := range pod.Status.Conditions {
```

Comment:
you can replace this loop with `IsStatusConditionTrue(pod.Status.Conditions, corev1.PodReady)` from pkg "k8s.io/apimachinery/pkg/api/meta"

Reply:
it does not seem to:
```
cannot convert pod.Status.Conditions (type []"k8s.io/api/core/v1".PodCondition) to type []"k8s.io/apimachinery/pkg/apis/meta/v1".Condition
```